src/main/java/tatracker/model/ModelManager.java

```diff
@@ -132,94 +157,125 @@ public void addModule(Module module) {
+        return taTracker.hasStudent(student);
     }
 
-    //=========== Session List Methods ================================================================================
+    // ======== Student Methods ================================================
 
     @Override
-    public boolean hasSession(Session session) {
-        requireNonNull(session);
-        return taTracker.hasSession(session);
+    public boolean hasStudent(Student student, Group targetGroup, Module targetModule) {
+        requireNonNull(student);
+        return taTracker.hasStudent(student);
     }
 
     @Override
-    public void addSession(Session session) {
-        taTracker.addSession(session);
-        updateFilteredSessionList(PREDICATE_SHOW_ALL_SESSIONS);
+    public void addStudent(Student student) {
+        taTracker.addStudent(student);
+        updateFilteredStudentList(PREDICATE_SHOW_ALL_STUDENTS);
     }
 
     @Override
-    public void deleteSession(Session target) {
-        taTracker.removeSession(target);
+    public void addStudent(Student student, Group targetGroup, Module targetModule) {
+        requireNonNull(student);
+        taTracker.addStudent(student);
+        updateFilteredStudentList(PREDICATE_SHOW_ALL_STUDENTS);
     }
 
     @Override
-    public void setSession(Session target, Session editedSession) {
-        requireAllNonNull(target, editedSession);
+    public void deleteStudent(Student target) {
+        taTracker.removeStudent(target);
+    }
 
-        taTracker.setSession(target, editedSession);
+    @Override
+    public void deleteStudent(Student target, Group targetGroup, Module targetModule) {
+        requireNonNull(target);
+        taTracker.removeStudent(target);
     }
 
-    //=========== Filtered Session List Accessors =============================================================
+    @Override
+    public void setStudent(Student target, Student editedStudent) {
+        requireAllNonNull(target, editedStudent);
+        taTracker.setStudent(target, editedStudent);
+    }
+
+    @Override
+    public void setStudent(Student target, Student editedStudent, Group targetGroup, Module targetModule) {
+        requireAllNonNull(target, editedStudent);
+        taTracker.setStudent(target, editedStudent);
+    }
 
     /**
-     * Returns an unmodifiable view of the list of {@code Session} backed by the internal list of
-     * {@code versionedTaTracker}
+     * TODO: Review filter functions.
      */
+
     @Override
-    public ObservableList<Session> getFilteredSessionList() {
-        return filteredSessions;
+    public ObservableList<Student> getFilteredStudentList() {
+        return filteredStudents;
     }
 
     @Override
-    public void updateFilteredSessionList(Predicate<Session> predicate) {
+    public void updateFilteredStudentList(Predicate<Student> predicate) {
         requireNonNull(predicate);
-        filteredSessions.setPredicate(predicate);
+        filteredStudents.setPredicate(predicate);
```

Comment:
Ok so I was thinking of the implementation of filter in relation to the module view and this is what I came up with:

- tatracker has a list of filtered modules
- modules has a list of filtered groups
- groups has a list of filtered students

This would fulfill all our purposes and be easy to implement without changing the current class structure. I will work on this with Yi Jing tomorrow first thing in the morning!